codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala

```diff
@@ -82,6 +82,85 @@ abstract class ModelBuilderSuite(val config: ModelBuilderSuite.Config) extends m
+      testFilesPath.resolve(config.descriptorSetPath + "/event-sourced-shoppingcart-with-java-pkg.desc")
+
+    Using(new FileInputStream(descriptorFilePath.toFile)) { fis =>
+      val fileDescSet = FileDescriptorSet.parseFrom(fis, config.registry)
+      val fileList = fileDescSet.getFileList.asScala
+
+      val descriptors: mutable.Seq[Descriptors.FileDescriptor] =
+        fileList.map(Descriptors.FileDescriptor.buildFrom(_, Array.empty, true))
+
+      val model = ModelBuilder.introspectProtobufClasses(descriptors)
+
+      val shoppingCartPackage =
+        PackageNaming(
+          "com/example/shoppingcart/shoppingcart_api.proto",
+          "ShoppingcartApi",
+          "example.shoppingcart",
+          Some("com.example.shoppingcart"),
+          Some("ShoppingCartApi"),
+          javaMultipleFiles = false)
+
+      val domainPackage =
+        PackageNaming(
+          "com/example/shoppingcart/domain/shoppingcart_domain.proto",
+          "ShoppingcartDomain",
+          "example.shoppingcart.domain",
+          Some("com.example.shoppingcart.domain"),
+          Some("ShoppingCartDomain"),
+          javaMultipleFiles = false)
+
+      val googleEmptyPackage =
+        PackageNaming(
+          "Empty",
+          "Empty",
+          "google.protobuf",
+          Some("com.google.protobuf"),
+          Some("EmptyProto"),
+          javaMultipleFiles = true)
+
+      val derivedEntityPackage = derivePackageForEntity(shoppingCartPackage, domainPackage)
+
+      val entity =
+        ModelBuilder.EventSourcedEntity(
+          // this is the name as defined in the proto file
+          fullyQualifiedName("ShoppingCart", derivedEntityPackage),
+          "shopping-cart",
+          ModelBuilder.State(fullyQualifiedName("Cart", domainPackage)),
+          List(
+            ModelBuilder.Event(fullyQualifiedName("ItemAdded", domainPackage)),
+            ModelBuilder.Event(fullyQualifiedName("ItemRemoved", domainPackage))))
+
+      assertEquals(model.entities, Map(entity.fqn.fullyQualifiedProtoName -> entity))
+
+//      assertEquals(
+//        model.services,
+//        Map(
+//          "example.shoppingcart.ShoppingCartService" ->
+//          ModelBuilder.EntityService(
+//            fullyQualifiedName("ShoppingCartService", shoppingCartPackage),
+//            List(
+//              command(
+//                "AddItem",
+//                fullyQualifiedName("AddLineItem", shoppingCartPackage),
+//                fullyQualifiedName("Empty", googleEmptyPackage)),
+//              command(
+//                "RemoveItem",
+//                fullyQualifiedName("RemoveLineItem", shoppingCartPackage),
+//                fullyQualifiedName("Empty", googleEmptyPackage)),
+//              command(
+//                "GetCart",
+//                fullyQualifiedName("GetShoppingCart", shoppingCartPackage),
+//                fullyQualifiedName("Cart", shoppingCartPackage))),
+//            entity.fqn.fullyQualifiedJavaName)))
+    }.get
```

Comment:
Oeps, forgot to re-enable this.

Reply:
Or maybe we should just remove this AST tests. 

It was useful to find out the issue and fix it, but we have other tests that cover the real generation and compilation. 

It's hard to read the 'expected' AST and even harder to know if the AST will effectively produce valid code.  At the end, we shape it to confirm what is emitted and that's not really testing anything. It's kind of tricking.

Reply:
Removed them, as I argued in https://github.com/lightbend/akkaserverless-java-sdk/issues/781, the AST tests have the highest maintenance overhead with the lowest value. 

It still has some value for Scala because we don't have ExampleSuite tests for it.